Infrastructure/Resources/New-ServiceBusConnectionString.ps1

```diff
@@ -0,0 +1,87 @@
+<#
+
+.SYNOPSIS
+Get a the connection string for a service bus namespace
+
+.DESCRIPTION
+Get a the connection string for a service bus namespace. If one does not exist with the given name, it will be created
+
+.PARAMETER ResourceGroupName
+The name of the destination Resource Group for the resource
+
+.PARAMETER NamespaceName
+The name of the Service Bus Namespace
+
+.PARAMETER AuthorizationRuleName
+The name of the authorization rule to be created
+
+.PARAMETER Rights
+A list of rights. This can be one or all of the following:
+Listen
+Send
+Manage
+
+.PARAMETER ConnectionStringType
+The connection string to return. This can be either Primary or Secondary.
+
+.EXAMPLE
+.\New-ServiceBusConnectionString.ps1 -NamespaceName test-namespace -AuthorizationRuleName auth1 -Rights Listen, Send, Manage -ConnectionStringType Secondary
+
+#>
+
+[CmdletBinding()]
+Param (
+    [Parameter(Mandatory = $true)]
+    [String]$NamespaceName,
+    [Parameter(Mandatory = $true)]
+    [String]$AuthorizationRuleName,
+    [Parameter(Mandatory = $true)]
+    [ValidateSet("Listen", "Send", "Manage")]
+    [String[]]$Rights,
+    [Parameter(Mandatory = $false)]
+    [ValidateSet("Primary", "Secondary")]
+    [String[]]$ConnectionStringType = "Primary"
+)
+
+try {
+    # --- Import Azure Helpers
+    Import-Module (Resolve-Path -Path $PSScriptRoot\..\Modules\Helpers.psm1).Path
+
+    Write-Log -LogLevel Information -Message "Checking for existing Service Bus Namespace: $NamespaceName"
+    $ServiceBusResource = Get-AzureRMResource -Name $NamespaceName
+    if (!$ServiceBusResource) {
+        throw "Could not find servicebus namespace $NamespaceName"
+    }
+
+    $AuthorizationRule = Get-AzureRmServiceBusAuthorizationRule -ResourceGroupName $ServiceBusResource.ResourceGroupName -Namespace $NameSpaceName -Name $AuthorizationRuleName -ErrorAction SilentlyContinue
+    if (!$AuthorizationRule) {
+        Write-Log -LogLevel Information -Message "Creating authorization rule with name $AuthorizationRuleName"
+        $AuthorizationRuleParameters = @{
+            ResourceGroupName = $ServiceBusResource.ResourceGroupName
+            Namespace         = $NamespaceName
+            Name              = $AuthorizationRuleName
+            Rights            = $Rights
+        }
+        $null = New-AzureRmServiceBusAuthorizationRule @AuthorizationRuleParameters
```

Comment:
Should this be expanded to be more granular so connection strings can be generated for a specific Queue or Topic? 
https://docs.microsoft.com/en-us/powershell/module/azurerm.servicebus/new-azurermservicebusauthorizationrule?view=azurermps-6.9.0